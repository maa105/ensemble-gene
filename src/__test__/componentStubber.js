import _ from 'lodash';

var lifecycleMethods = [
  'render',
  'componentWillMount',
  'componentDidMount',
  'componentWillReceiveProps',
  'shouldComponentUpdate',
  'componentWillUpdate',
  'componentDidUpdate',
  'componentWillUnmount'
];

var stubComponent = function(componentClass) {
  var originalPropTypes, mocked = [];

  beforeEach(function() {
    originalPropTypes = componentClass.propTypes;

    componentClass.propTypes = {};

    _.forEach(lifecycleMethods, function(method) {
      if(typeof componentClass.prototype[method] !== 'undefined') {
        mocked.push(method);
        const spy = jest.spyOn(componentClass.prototype, method).mockImplementation(() => null);
        componentClass.prototype[method].spy = spy;
      }
    });
  });

  afterEach(function() {
    componentClass.propTypes = originalPropTypes;
    _.forEach(mocked, function(method) {
      componentClass.prototype[method].spy.mockRestore();
    });
  });
};

module.exports = {
  stubComponent: stubComponent
};
