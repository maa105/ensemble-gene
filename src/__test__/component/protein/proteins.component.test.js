import React from 'react';
import Protein from '../../../components/protein/protein.component';
import Proteins from '../../../components/protein/proteins.component';
import renderer from 'react-test-renderer';
import { stubComponent } from '../../componentStubber';

describe('protien component test', function() {
  stubComponent(Protein);
  
  const props = { 
    proteins: [
      { id: 'protein1', seq: 'seq1' },
      { id: 'protein2', seq: 'seq2' }
    ],
    position: 600 
  };

  test('Proteins with proper props to render correctly', () => {
    const component = renderer.create(
      <Proteins {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const proteins = component.root.findAllByType(Protein);
    expect(proteins.length).toBe(2);
    
    const protein1Props = proteins[0].props;
    const protein2Props = proteins[1].props;

    // props passed down properly to Protein 
    expect(protein1Props).toEqual({
      bsStyle: undefined, // from react bootstrap
      eventKey: 0,
      protein: { id: 'protein1', seq: 'seq1' },
      position: 600
    });
    expect(protein2Props).toEqual({
      bsStyle: undefined, // from react bootstrap
      eventKey: 1,
      protein: { id: 'protein2', seq: 'seq2' },
      position: 600
    });

  });
});
