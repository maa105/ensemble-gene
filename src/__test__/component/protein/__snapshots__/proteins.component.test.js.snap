// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protien component test Proteins with proper props to render correctly 1`] = `
<div
  className="panel-group"
  id="proteins"
  role="tablist"
/>
`;
